docs/terms/metric.md

```diff
@@ -4,12 +4,104 @@ title: Metric
+
+### Past Metrics are available to other Expectations and Data Docs
+
+An Expectation can also expose Metrics, such as the observed value of a useful statistic via an Expectation Validation Result, where <TechnicalTag relative="../" tag="data_docs" text="Data Docs" /> -- or other Expectations -- can use them.  This is done through an Action (to which the Expectation's Validation Result has been passed) which will save them to a Metric Store.  The Action in question is the `StoreMetricsAction`.  You can view the implementation of this Action in the `great_expectations/checkpoint/actions.py` file of a Great Expectations deployment, or the [same file in our GitHub.](https://github.com/great-expectations/great_expectations/blob/0312642755f6003c70623e9aa3ceed1020373dac/great_expectations/checkpoint/actions.py#L905)
+
+## API basics
+
+### How to access
+
+Validation Results can expose Metrics that are defined by specific Expectations that have been validated, called "Expectation Defined Metrics." To access those values, you address the Metric as a dot-delimited string that identifies the value, such as `expect_column_values_to_be_unique.success`or `expect_column_values_to_be_between.result.unexpected_percent`. These Metrics may be stored in a Metrics Store.
+
+It is important to note that these Metrics are *NOT* available for compute on raw data directly as other Metrics are.
+
+A `metric_kwargs_id` is a string representation of the Metric Kwargs that can be used as a database key. For simple cases, it could be easily readable, such as `column=Age`, but when there are multiple keys and values or complex values, it will most likely be a md5 hash of key/value pairs. It can also be `None` in the case that there are no kwargs required to identify the Metric.
+
+The following examples demonstrate how Metrics are defined:
+
+```python title="Python code"
+res = df.expect_column_values_to_be_in_set(
+    "color",
+    ["red", "green"]
+)
+res.get_metric(
+    "expect_column_values_to_be_in_set.result.missing_count",
+    column="color"
+)
+```
+
+See the [How to configure a MetricsStore](../guides/setup/configuring_metadata_stores/how_to_configure_a_metricsstore.md) guide for more information.
+
+### How to create
+
+Metrics are produced using logic specific to the Execution Engine associated with the Datasource that provides the data for the Batch Request/s that the Metric is calculated for.  That logic that is defined in a `MetricProvider`. When a `MetricProvider` class is first encountered, Great Expectations will register the Metric and any methods that it defines as able to produce Metrics.  This means that after Validation, that Metric can then be referenced in Evaluation Parameters. Furthermore, the methods associated with a Metric can also be referenced through dot notation using the Metric as a base (as shown above in the [section on accessing Metrics](#how-to-access)). 
```

Comment:
I don't think this is quite right. I don't think the reference to Evaluation Parameters should be here. Evaluation Parameters can be used to get an already calculated Metric, whereas registration is more for the method of Metric calculation. The registered metric will then be able to be used with `validator.get_metric()` / `validator.get_metrics()`.

Reply:
I see.  I misunderstood some of the documentation I was collating for this, then.  I've updated the section accordingly.